tests/ui/lifetimes/shorter-tail-expr-lifetime.rs

```diff
@@ -0,0 +1,15 @@
+//@ revisions: edition2021 edition2024
+//@ [edition2024] compile-flags: -Zunstable-options
+//@ [edition2024] edition: 2024
+//@ [edition2024] run-pass
+
+#![cfg_attr(edition2024, feature(shorter_tail_lifetimes))]
+
+fn f() -> usize {
+    let c = std::cell::RefCell::new("..");
+    c.borrow().len() //[edition2021]~ ERROR: `c` does not live long enough
+}
+
+fn main() {
+    let _ = f();
+}
```

Comment:
Many tests should be added. Such as
1. code will still panic, but leave the mutex unpoisoned
```rust
fn f(m: &Mutex<i32>) -> i32 {
    let _x = PanicOnDrop;
    *m.lock().unwrap()
}
```
2. mutable reference
```
fn main() {
    let c = std::cell::RefCell::new(123);

    if let Ok(mut b) = c.try_borrow_mut() {
        *b = 321;
    };
}
```

3. Explicit drop order
https://github.com/rust-lang/rfcs/pull/3606/files#r1548272644

4. injects a borrow to a temporary into an already existing local variable that borrows it on drop
```
fn why_would_you_do_this() -> bool {
    let mut x = None;
    // Make a temporary `RefCell` and put a `Ref` that borrows it in `x`.
    x.replace(RefCell::new(123).borrow()).is_some()
}
```
5. block could be a subexpression of a larger expression
```
    let zero = { String::new().as_str() }.len();
```
Considering this is a breaking feature, more tests would be better.

Reply:
Thanks for the suggestions! Some additional tests have been checked in.